### ISSUE TITLE
NEW should report error on missing parameters

### ISSUE THREAD

**Issue by larshp** [abaplint/abaplint#2154]
```abap
CLASS lcl_stream DEFINITION.
  PUBLIC SECTION.
    METHODS constructor IMPORTING iv_str TYPE string.
    METHODS take_matching RETURNING VALUE(ro_stream) TYPE REF TO lcl_stream.
ENDCLASS.

CLASS lcl_stream IMPLEMENTATION.
  METHOD constructor.
  ENDMETHOD.

  METHOD take_matching.
    ro_stream = NEW #( ).
  ENDMETHOD.
ENDCLASS.
```
